src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverterFactory.cs

```diff
@@ -36,6 +36,7 @@ internal static JsonConverter Create(Type enumType, EnumConverterOptions convert
         [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2055:MakeGenericType",
             Justification = "'EnumConverter<T> where T : struct' implies 'T : new()', so the trimmer is warning calling MakeGenericType here because enumType's constructors are not annotated. " +
             "But EnumConverter doesn't call new T(), so this is safe.")]
+        [UnconditionalSuppressMessage("AOT", "IL3050:Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.", Justification = "<Pending>")]
```

Comment:
Enum types are valuetypes, the MakeGenericType will likely throw 100% of time.

Reply:
I suspect that the original justification is misleading, it is safe to suppress factory linker warnings because we're only rooting them in [reflection-based serialization APIs](https://github.com/dotnet/runtime/blob/6c7ce85e149eaaf4cc452a7008754d5c799db938/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs#L39). We might want to annotate the factory constructors as `RequiresUnreferencedCode`, including the public [`JsonStringEnumConverter`](https://github.com/dotnet/runtime/blob/6c7ce85e149eaaf4cc452a7008754d5c799db938/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonStringEnumConverter.cs#L55).

cc @eerhardt

Reply:
You might consider suppressing the warning here and add an annotation in the converter factory constructor, as is done here: 

https://github.com/dotnet/runtime/blob/765ecde09bff36604a32bf597b706d6a59155723/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IEnumerableConverterFactory.cs#L31-L32

The factory constructors are only invoked in the context of [reflection serialization code paths](https://github.com/dotnet/runtime/blob/6c7ce85e149eaaf4cc452a7008754d5c799db938/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs#L39), with the notable exception of [`JsonStringEnumConverter`](https://github.com/dotnet/runtime/blob/6c7ce85e149eaaf4cc452a7008754d5c799db938/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonStringEnumConverter.cs) which is a public class. It would likely require an annotation on its constructors.